### PR TITLE
Honor `-c` with `rgbgfx -r`

### LINKED ISSUE #1166: `rgbgfx -r` ignores `-c`
I tested with an inline palette spec, and still got a grayscale image as output. No attributes were passed, either.

### PR DESCRIPTION
**Pull request by ISSOtm**
Fixes #1166

### CODE REVIEW

File: src/gfx/reverse.cpp [hunk 1]
```diff
@@ -141,6 +141,7 @@ void reverse() {
 	std::vector<std::array<Rgba, 4>> palettes{
 	    {Rgba(0xFFFFFFFF), Rgba(0xAAAAAAFF), Rgba(0x555555FF), Rgba(0x000000FF)}
     };
+    // If a palette file is used as input, it overrides the default colours.
```
Comment (Rangi42) [inline src/gfx/reverse.cpp:144]: :pick: Tab indent
```suggestion
	};
	// If a palette file is used as input, it overrides the default colours.
```

Comment (ISSOtm) [reply src/gfx/reverse.cpp]: Weird.

File: src/gfx/reverse.cpp [hunk 2]
```diff
@@ -171,6 +172,14 @@ void reverse() {
 			warning("Read %zu palettes, more than the specified limit of %zu", palettes.size(),
 			        options.nbPalettes);
 		}
+
+		if (options.palSpecType == Options::EXPLICIT && palettes != options.palSpec) {
+			warning("Colors in the palette file do not match those specified with `-c`!\n");
```
Comment (Rangi42) [inline src/gfx/reverse.cpp:177]:
```suggestion
			warning("Colors in the palette file do not match those specified with `-c`!");
```

File: src/gfx/reverse.cpp [hunk 3]
```diff
@@ -171,6 +172,14 @@ void reverse() {
...
+		}
+	} else if (options.palSpecType == Options::EMBEDDED) {
+		warning("An embedded palette was requested, but no palette file was specified; ignoring request.\n");
```
Comment (Rangi42) [inline src/gfx/reverse.cpp:180]:
```suggestion
		warning("An embedded palette was requested, but no palette file was specified; ignoring request.");
```

**Review by Rangi42 — APPROVED** [review]
Tested with `-c` alone and with `-c -p`; works as expected.

**Pull request merged**